.github/workflows/partial-tests.yml

```diff
@@ -128,7 +138,17 @@ jobs:
             core:
               - 'taipy/core/**'
 
-      - uses: actions/setup-python@v5
+      - name: Setup Python with cache
+        if: steps.changes.outputs.core == 'true'
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{matrix.python-version}}
+          cache: 'pipenv'
+          cache-dependency-path: '**/Pipfile'
+
+      - name: Setup Python without cache
+        if: steps.changes.outputs.core != 'true'
+        uses: actions/setup-python@v5
         with:
           python-version: ${{matrix.python-version}}
```

Comment:
The `submit_tests` job is only run if some core files are touched. This step is run when core files are not touched. I don't get its purpose.